src/java/htsjdk/samtools/SRAFileReader.java

```diff
@@ -0,0 +1,275 @@
+     * @return An iterator over the given chunks.
+     */
+    @Override
+    public SAMRecordIterator iterator(final SAMFileSpan chunks) {
+        CloseableIterator<SAMRecord> it = getIterator(chunks);
+        if (it == null) {
+            return null;
+        }
+        return (SAMRecordIterator) it;
+    }
+
+    /** ReaderImplementation */
+    @Override
+    void enableFileSource(final SamReader reader, final boolean enabled) {
+        log.info("enableFileSource is not supported");
+    }
+
+    @Override
+    void enableIndexCaching(final boolean enabled) {
+        log.info("enableIndexCaching is not supported");
+    }
+
+    @Override
+    void enableIndexMemoryMapping(final boolean enabled) {
+        log.info("enableIndexMemoryMapping is not supported");
+    }
+
+    @Override
+    void enableCrcChecking(final boolean enabled) {
+        log.info("enableCrcChecking is not supported");
+    }
+
+    @Override
+    void setSAMRecordFactory(final SAMRecordFactory factory) {
+        log.info("setSAMRecordFactory is not supported");
+    }
+
+    @Override
+    void setValidationStringency(final ValidationStringency validationStringency) {
+        this.validationStringency = validationStringency;
+    }
+
+    protected SRAIterator.RecordRangleInfo getRecordsRangeInfo() {
+        return recordRangeInfo;
+    }
+
+    private SAMFileHeader loadSamHeader() throws ErrorMsg {
+        if (run == null) {
+            throw new RuntimeException("Cannot load SAMFileHeader - SRA run is uninitialized");
+        }
+
+        String runName = run.getName();
+
+        SAMFileHeader header = new SAMFileHeader();
+        header.setSortOrder(SAMFileHeader.SortOrder.coordinate);
+
+        ReadGroupIterator itRg = run.getReadGroups();
+        while (itRg.nextReadGroup()) {
+            String rgName = itRg.getName();
+            if (rgName.equals(""))
+                rgName = "_";
+            SAMReadGroupRecord rg = new SAMReadGroupRecord(rgName);
+            rg.setSample(runName);
+            header.addReadGroup(rg);
+        }
+
+        ReferenceIterator itRef = run.getReferences();
+        while (itRef.nextReference()) {
+            header.addSequence(new SAMSequenceRecord(itRef.getCanonicalName(), (int) itRef.getLength()));
+        }
+
+        return header;
+    }
+
+    private List<Reference> loadReferences() throws ErrorMsg {
+        if (run == null) {
+            throw new RuntimeException("Cannot cache references - SRA run is uninitialized");
+        }
+        if (virtualHeader == null) {
+            throw new RuntimeException("Cannot cache references - header is uninitialized");
+        }
+
+        SAMSequenceDictionary sequenceDictionary = virtualHeader.getSequenceDictionary();
+        List<Reference> references = new ArrayList<Reference>(sequenceDictionary.size());
+        for (SAMSequenceRecord sequence : sequenceDictionary.getSequences()) {
+            references.add(null);
```

Comment:
Given that this method does nothing except add a bunch of `null`s to a `List`, it seems inappropriate to call it `loadReferences()`. Was this your intent? Is the `cachedReferences` member even used? (it does not appear to be)